### PR TITLE
User edit

### PR DESCRIPTION
**Pull request by jonzee**

### CODE REVIEW

File: src/ts/Menu/index.tsx [hunk 1]
```diff
@@ -15,28 +17,30 @@ export default function MenuVG(props) {
...
       anchorOrigin={{ horizontal: 'left', vertical: 'bottom' }}
       desktop={true}
     >
-      <MenuItem primaryText={i('Profile')} href="/profile" />
+      <MenuItem primaryText={i('Profile')} href="/user/profile" />
       <MenuItem primaryText={i('Add offer')} href="/add-offer" />
       <MenuItem primaryText={i('Deals')} href="/deals" />
-      <MenuItem primaryText={i('Messages')} href="/messages" />
-      <MenuItem primaryText={i('Account')} href="/account" />
+      <MenuItem primaryText={i('Messages')} href="user/inbox" />
```
Comment (michalmikolajczyk) [inline src/ts/Menu/index.tsx]: leading dash?

File: src/ts/Menu/index.tsx [hunk 2]
```diff
@@ -15,28 +17,30 @@ export default function MenuVG(props) {
...
+      <MenuItem primaryText={i('Account')} href="user/account" />
```
Comment (michalmikolajczyk) [inline src/ts/Menu/index.tsx]: as above

File: src/ts/NotFound/index.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import Container from '../Container';
 export default function NotFound(props) {
 	return (
     <Container>
-      <div className="cont">
+      <div className="cont" style={{height: 500, textAlign: 'center', lineHeight: '200px'}}>
```
Comment (michalmikolajczyk) [inline src/ts/NotFound/index.tsx]: ?

File: src/ts/Profile/UserOptional.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import { FormsyText } from 'formsy-material-ui/lib';
+import i from '../i18n';
+
+export default function UserOptional(props) {
+  const { user } = props;
+  const disabled = !user;
+  return (
+    <Paper className="user-form">
+      <div className="head">
+        {i('Optional Information')}
```
Comment (michalmikolajczyk) [inline src/ts/Profile/UserOptional.tsx]: let's discuss the way we load translations

File: src/ts/Profile/index.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+    expect(instance.setState).to.not.have.been.called;
+    expect(instance.state.user).to.be.null;
+    expect(instance.state.networkErr).to.be.false;
+
+    instance['componentDidMount']();
+    get.rejects('unexpected error');
+
+    expect(instance.setState).to.have.been.calledOnce;
+    expect(instance.state.networkErr).to.be.true;
```
Comment (michalmikolajczyk) [inline src/ts/Profile/index.spec.tsx]: :+1:

File: src/ts/Signin/SigninForm.tsx [hunk 6]
```diff
@@ -42,10 +42,18 @@ export default class SiginForm extends React.Component<any, any> {
...
       >
         <FormsyText
-          name="name"
+          name="firstname"
           required={true}
           fullWidth={true}
-          floatingLabelText={i('Full Name')}
+          floatingLabelText={i('First Name')}
+          validations="minLength:3"
+          validationError={i('Full name too short! Minimum 3 chars')}
```
Comment (michalmikolajczyk) [inline src/ts/Signin/SigninForm.tsx]: relax with the copy here. please kindly ask the user to add at least three characters

File: src/ts/i18n/pl.ts [hunk 7]
```diff
@@ -19,6 +19,24 @@ export default {
   'FAQ': 'FAQ',
 
 /**
+ * Dialogs
+ */
+// Network Error
+  // 'Cancel': '',
+  'Network Error': 'Błąd Połączenia',
+  'Ups! It seems that we have some problems with network, try again later.':
+  'Ups! Wygląda na to, że mamy problemy z siecią, spróbuj ponownie później.',
```
Comment (michalmikolajczyk) [inline src/ts/i18n/pl.ts]: copy

File: src/ts/i18n/pl.ts [hunk 8]
```diff
@@ -19,6 +19,24 @@ export default {
...
+
+// Save Error
+  'Save Error': 'Błąd Zapisu',
+  'There were some problems with saving new data. Please check whether the data are correct and try again':
+  'Pojawił się błąd przy zapisie danych. Sprawdź proszę czy wprowadzone dane są poprawne i spróbuj ponownie',
```
Comment (michalmikolajczyk) [inline src/ts/i18n/pl.ts]: copy

**Comment by michalmikolajczyk** [comment]
pomniejsze uwagi, generalnie :pizza: dobra robota

File: src/ts/NotFound/index.tsx (hunk 3 above)
Comment (jonzee) [inline src/ts/NotFound/index.tsx]: user-edit profile has many new routes, so I fast-improved 404 page

File: src/ts/Profile/UserOptional.tsx (hunk 4 above)
Comment (jonzee) [inline src/ts/Profile/UserOptional.tsx]: there is a map english-string to polish-string and module that checks browser language and chooses right translation (checkout `src/ts/i18n`)

File: src/ts/Signin/SigninForm.tsx (hunk 6 above)
Comment (jonzee) [inline src/ts/Signin/SigninForm.tsx]: ok, I think that we can discuss the copy with our team

File: src/ts/NotFound/index.tsx (hunk 3 above)
Comment (michalmikolajczyk) [inline src/ts/NotFound/index.tsx]: let's put that into some css file then

File: src/ts/Profile/UserOptional.tsx (hunk 4 above)
Comment (michalmikolajczyk) [inline src/ts/Profile/UserOptional.tsx]: :+1:

**Pull request merged**